node/sc/local_backend.go

```diff
@@ -66,277 +43,6 @@ func checkCtx(ctx context.Context) error {
-		evm.Cancel(vm.CancelByCtxDone)
-	}()
-
-	result, err := blockchain.ApplyMessage(evm, msg)
-	if err := vmError(); err != nil {
-		return nil, err
-	}
-
-	if err != nil {
-		return nil, err
-	}
-
-	return result.Return(), result.Unwrap()
-}
-
-func (lb *LocalBackend) PendingCodeAt(ctx context.Context, contract common.Address) ([]byte, error) {
-	if err := checkCtx(ctx); err != nil {
-		return nil, err
-	}
-	// TODO-Klaytn this is not pending code but latest code
-	return lb.CodeAt(ctx, contract, lb.subbridge.blockchain.CurrentBlock().Number())
-}
-
-func (lb *LocalBackend) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
-	if err := checkCtx(ctx); err != nil {
-		return 0, err
-	}
-	return lb.subbridge.txPool.GetPendingNonce(account), nil
-}
-
-func (lb *LocalBackend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
-	if err := checkCtx(ctx); err != nil {
-		return nil, err
-	}
-	if lb.subbridge.blockchain.Config().IsMagmaForkEnabled(lb.subbridge.blockchain.CurrentHeader().Number) {
-		return new(big.Int).SetUint64(lb.config.Governance.KIP71.UpperBoundBaseFee), nil
-	} else {
-		return new(big.Int).SetUint64(lb.config.UnitPrice), nil
-	}
-}
-
-func (lb *LocalBackend) EstimateGas(ctx context.Context, call klaytn.CallMsg) (gas uint64, err error) {
-	if err := checkCtx(ctx); err != nil {
-		return 0, err
-	}
-	// Binary search the gas requirement, as it may be higher than the amount used
-	var (
-		lo  uint64 = params.TxGas - 1
-		hi  uint64
-		cap uint64
-	)
-	if uint64(call.Gas) >= params.TxGas {
-		hi = uint64(call.Gas)
-	} else {
-		hi = params.UpperGasLimit
-	}
-	cap = hi
-
-	// Create a helper to check if a gas allowance results in an executable transaction
-	executable := func(gas uint64) bool {
-		call.Gas = gas
-
-		currentState, err := lb.subbridge.blockchain.State()
-		if err != nil {
-			return false
-		}
-		if _, err := lb.callContract(ctx, call, lb.subbridge.blockchain.CurrentBlock(), currentState); err != nil {
-			return false
-		}
-		return true
-	}
-	// Execute the binary search and hone in on an executable gas limit
-	for lo+1 < hi {
-		mid := (hi + lo) / 2
-		if !executable(mid) {
-			lo = mid
-		} else {
-			hi = mid
-		}
-	}
-	// Reject the transaction as invalid if it still fails at the highest allowance
-	if hi == cap {
-		if !executable(hi) {
-			return 0, fmt.Errorf("gas required exceeds allowance or always failing transaction")
-		}
-	}
-	return hi, nil
-}
-
-func (lb *LocalBackend) SendTransaction(ctx context.Context, tx *types.Transaction) error {
-	if err := checkCtx(ctx); err != nil {
-		return err
-	}
-	return lb.subbridge.txPool.AddLocal(tx)
-}
-
-// ChainID can return the chain ID of the chain.
-func (lb *LocalBackend) ChainID(ctx context.Context) (*big.Int, error) {
-	if err := checkCtx(ctx); err != nil {
-		return nil, err
-	}
-	return lb.config.ChainID, nil
-}
-
-func (lb *LocalBackend) TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
```

Comment:
Why was the decision made to omit the migration of this function? Its absence could potentially lead to a crash during bridge deployment, which wasn't addressed in the unit tests.

Reply:
Because `bind.ContractTransactor` does not include `TransactionReceipt` and I tested value transfer across bridge without `TransactionReceipt` which worked fine.

Also, adding `TransactionReceipt` does not look trivial because `BlockChainForCaller` provides a limited set of functions.. I'll have to manually iterate block 0 ~ current and find if corresponding transaction exists.

Reply:
There are two ways to bridge contract deployments.
1.  By contract deployment transaction through web3, ethers, caver, etc.
2. `subbridge.deployBridge()` RPC

There's no side effect with the first approach. The second one also works but it has different behavior internally.
As the localbackend does not have `TransactionReceipt` method, which is deleted in this PR, the [type assertion](https://github.com/klaytn/klaytn/blob/dev/node/sc/bridge_manager.go#L939C1-L943C3) fails here and do not call `WaitDeployed()`. Thus, I suggest that the refactoring should refrain from any underlying behaviors to be modified implicitly. 

>  I'll have to manually iterate block 0 ~ current and find if corresponding transaction exists.

The following code snippet works. I'll look into this further in detail later if you confirm.
```go
func (b *BlockchainContractCaller) TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
    bc := b.bc.(*blockchain.BlockChain) // Avoid including a conversion check at this point, as the BlockChain struct conversion must be done. Additionally, there is no anticipation of encountering errors in this function.
    receipt := bc.GetReceiptByTxHash(txHash)
    if receipt != nil {
        return receipt, nil
    }
    return nil, errors.New("receipt does not exist")
}
```

Reply:
I agree with @hyunsooda that the `BlockChainContractBackend` should implement `DeployBackend`. `WaitDeployed` and `WaitMined` is such a useful feature.

I understand that `BlockChainForCaller` has only a limited set of methods. However, the concrete instance will be `*blockchain.BlockChain` in most cases, if not all. So `b.bc.(*blockchain.BlockChain)` should work fine (at least when we use `WaitDeployed` and such)

The following method might look more panic-safe.
```go
    if bc, ok := b.bc.(*blockchain.BlockChain); ok {
    	// return receipt or error
    } else {
        return nil, errors.New("Not BlockChain")
    }
```

Reply:
@hyunsooda @blukat29 Thanks for the feedbacks, PTAL at a5d598c.